app/vmctl/vm_native.go

```diff
@@ -274,3 +384,81 @@ func (c *vmNativeClient) do(req *http.Request, expSC int) (*http.Response, error
 	}
 	return resp, err
 }
+
+type LabelValues map[string]string
+
+type Response struct {
+	Status string        `json:"status"`
+	Series []LabelValues `json:"data"`
+}
+
+func (p *vmNativeProcessor) explore(ctx context.Context, f filter) (map[string]struct{}, error) {
+	url := fmt.Sprintf("%s/%s", p.src.addr, nativeSeriesAddr)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create request to %q: %s", url, err)
+	}
+
+	params := req.URL.Query()
+	if f.timeStart != "" {
+		params.Set("start", f.timeStart)
+	}
+	if f.timeEnd != "" {
+		params.Set("end", f.timeEnd)
+	}
+	params.Set("match[]", f.match)
+	req.URL.RawQuery = params.Encode()
+
+	resp, err := p.src.do(req, http.StatusOK)
+	if err != nil {
+		return nil, fmt.Errorf("tenants request failed: %s", err)
+	}
+
+	var response Response
+	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
+		return nil, fmt.Errorf("cannot decode tenants response: %s", err)
+	}
+
+	if err := resp.Body.Close(); err != nil {
+		return nil, fmt.Errorf("cannot close tenants response body: %s", err)
+	}
+	names := make(map[string]struct{})
+	for _, series := range response.Series {
+		for labelName, labelValue := range series {
+			if labelName == nameLabel {
+				name := fmt.Sprintf("{%s=%q}", nameLabel, labelValue)
```

Comment:
nit: would be better to store name as is. 
And then form a proper formatting on query time, since there will be an allocation anyway.

Reply:
updated logic